### PR TITLE
Allow `pragma(mangle,...)` on an aggregate

### PR DESCRIPTION
**Pull request by thewilsonator**
see https://github.com/dlang/dmd/pull/11835

### CODE REVIEW

File: spec/pragma.dd [hunk 1]
```diff
@@ -220,21 +220,39 @@ $(H3 $(LNAME2 mangle, $(D pragma mangle)))
     $(P Overrides the default mangling for a symbol.)
 
     $(P There must be one $(ASSIGNEXPRESSION) and it must evaluate at compile time to a string literal.
+        If the declaration this pragma is attached to is an aggregate there may be an additional
+        $(ASSIGNEXPRESSION) that must evaluate to a symbol. If that symbol is a TemplateInstance the aggregate
+        is treated as being a template with the signature and arguments of the TemplateInstance.
```
Comment (WalterBright) [inline spec/pragma.dd:225]: 224-225 make no sense to me.

File: spec/pragma.dd [hunk 2]
```diff
@@ -220,21 +220,39 @@ $(H3 $(LNAME2 mangle, $(D pragma mangle)))
...
     )
 -----------------
 pragma(mangle, "body")
 extern(C) void body_func();
+pragma(mangle, "function")
+extern(C++) struct _function {}
+
+template ScopeClass(C)
+{
+    pragma(mangle, C.stringof, C)
```
Comment (WalterBright) [inline spec/pragma.dd:248]: I don't see how the third argument changes anything?

File: spec/pragma.dd [hunk 3]
```diff
@@ -291,8 +295,27 @@ $(H3 $(LNAME2 mangle, $(D pragma mangle)))
...
+        }
         ---
     )
+=======
+-----------------
+
+
+
+-----------------
+>>>>>>> Allow `pragma(mangle,...)` on an aggregate
```
Comment (Geod24) [inline spec/pragma.dd:318]: ?

Comment (thewilsonator) [reply spec/pragma.dd]: accidentally committed half a merge conflict marker. fixed.

File: spec/pragma.dd [hunk 4]
```diff
@@ -272,7 +272,11 @@ $(H3 $(LNAME2 mangle, $(D pragma mangle)))
 
     $(P Overrides the default mangling for a symbol.)
 
-    $(P There must be one $(ASSIGNEXPRESSION) and it must evaluate at compile time to a string literal.
+    $(P For variables and functions there must be one $(ASSIGNEXPRESSION) and it must evaluate at compile time to a string literal.
+        For aggregates there may one or two $(ASSIGNEXPRESSION)s, one of which must evaluate at compile time to a string literal,
```
Comment (mdparker) [inline spec/pragma.dd:276]: "literal, " -> "literal and"

File: spec/pragma.dd [hunk 5]
```diff
@@ -272,7 +272,11 @@ $(H3 $(LNAME2 mangle, $(D pragma mangle)))
...
+        one which must evaluate to a symbol. If that symbol is a TemplateInstance the aggregate is treated as being a template
```
Comment (mdparker) [inline spec/pragma.dd:277]: "TemplateInstance" should be italicized, and needs a comma.

File: spec/pragma.dd [hunk 6]
```diff
@@ -272,7 +272,11 @@ $(H3 $(LNAME2 mangle, $(D pragma mangle)))
...
+        with the signature and arguments of the TemplateInstance. If no string is supplied the identifier of the symbol is used
```
Comment (mdparker) [inline spec/pragma.dd:278]: "TemplateInstance" should be italicized.

Comment (mdparker) [inline spec/pragma.dd:278]: "If no string is supplied the identifier of the symbol is used" -> "The identifier of the symbol is used when no string is supplied."

File: spec/pragma.dd (hunk 5 above)
Comment (mdparker) [inline spec/pragma.dd:277]: "treated as being a" -> "treated as a"

File: spec/pragma.dd (hunk 6 above)
Comment (mdparker) [inline spec/pragma.dd:278]: "with the" -> "that has the"

File: spec/pragma.dd [hunk 7]
```diff
@@ -272,7 +272,11 @@ $(H3 $(LNAME2 mangle, $(D pragma mangle)))
...
+        in its place. Both arguments may be used used when an aggregate has name that is a D keyword.
```
Comment (mdparker) [inline spec/pragma.dd:279]: "an aggregate has name that is" -> "an aggregate's name is"

**Review by mdparker — CHANGES_REQUESTED** [review]

**Comment by RazvanN7** [comment]
@thewilsonator please apply the feedback.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**